src/segment-loader.js

```diff
@@ -725,6 +727,133 @@ export default class SegmentLoader extends videojs.EventTarget {
   }
 
   /**
+   * Determines if the network has enough bandwidth to complete the current segment
+   * request in a timely manner. If not, the request will be aborted early and bandwidth
+   * updated to trigger a playlist switch.
+   *
+   * @param {Object} stats
+   *        Object continaing stats about the request timing and size
+   * @return {Boolean} True if the request was aborted, false otherwise
+   * @private
+   */
+  abortRequestEarly_(stats) {
+    if (this.hls_.tech_.paused() ||
```

Comment:
Anyway to break this almost 100 line function down into a couple of smaller functions?  

Maybe some of the "ifs" could be moved elsewhere? Or some of the logic in the sort functions?